Automattic/jetpack
# Map block: Add absolute positioning for container only when Mapkit is used

Fixes https://github.com/Automattic/jetpack/issues/39722

## Proposed changes:

* This PR moved absolute positioning out of a Map block stylesheet and ensures it's dynamically added if a map is using Mapkit. This is to prevent an issue with blocks placed underneath the map block overlapping with the block, if using Mapbox (affecting only self-hosted sites)

### Other information:


## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Self-hosted:

* On trunk or bleeding edge / current Jetpack version on a test site, add add a Map block to a post or page.
* Try to add a block underneath, notice that it overlaps with the Map block. If you click on the 'resizable box' handle - the small blue circle and corresponding horizontal line visible when hovering over the selected block - it will auto-size and fix the issue, so you'll want to avoid doing that just now - or add a new Map block.
* Apply this PR, and notice that the same 'resize handle' is now at the bottom of the block, and new blocks now no longer overlap.

WoA / Simple:

* There should be no change in behaviour on WoA / Simple. However the absolute positioning was originally introduced to ensure the map is visible for sites using Mapkit, which is WoA / Simple sites, so it would be good to test both platforms and make sure the maps continue to display as expected, and then when the blocks are selected the map settings are visible in the plugin sidebar as well. See this PR for original testing instructions: https://github.com/Automattic/jetpack/pull/38956
* On WoA, make sure to test trunk first, as you may otherwise encounter an unrelated issue if you were previously testing an older branch, as the Jetpack blocks toggle may need re-enabling (see p1728387435601079-slack-C02FMH4G8)
